src/ast/property.js

```diff
@@ -28,6 +29,7 @@ module.exports = Statement.extends(
     readonly,
     nullable,
     type,
+    hooks,
```

Comment:
It would probably make sense to put hooks last as this would be less breaking of a change for existing code. Also making the default [] would make the type signature simpler, and reduce ambiguity between an empty list and null, as an empty property list is not valid syntax.

Reply:
Adding the `hooks` parameter at the end of the list would be inconsistent with how things have been done so far.

The PR introducing support for typed class constant added type "in the middle". https://github.com/glayzzle/php-parser/pull/1136/files#diff-33d9ef64a624f7abad7378a94559475184322ac87b7c864326beea672657adbcR34

Reply:
@cseufert I do not think it is possible to put hooks as the last parameters: when we [create a node](https://github.com/glayzzle/php-parser/pull/1143/files#diff-fc169768643555846d9df28c56bdfeebd845afe68e2d045596b56ad7b915a9acR221-R229), the parameter `docs` and `location` are passed in by some other layer.

Reply:
@czosel any opinion on this?